ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,810 @@
+    my $excludes      = $t->{excludes};
+    print $yum_conf_file $t->{yumconf};
+    print $yum_conf_file "exclude=" . join ( " ", sort @$excludes );
+    $yum_conf_file->close();
+
+    if ( !$NoAction ) {
+        # Remove all repositories if userpkgs is not defined
+        $self->cleanup_old_repos( REPOS_DIR, $repos, $t->{userpkgs} ) or return 0;
+    }
+
+    # Generate new installation repositories from host profile
+    my $proxy = !defined($t->{proxy}) ? 'yes' : $t->{proxy};
+    foreach my $repo (@$repos) {
+        my $fh = CAF::FileWriter->new( REPOS_DIR . "/spma-$repo->{name}.repo", log => $self );
+        my $prots = $repo->{protocols}->[0];
+        my $url   = $prots->{url};
+        my $urls;
+        my $repo_proxy = defined($repo->{proxy}) ? 'yes' : 'no';
+        my $disable_proxy = defined($repo->{disableproxy}) ? 'yes' : 'no';
+        if ( $url =~ /http/ && $proxy eq 'yes' && $disable_proxy eq 'no' ) {
+            if ( $config->elementExists(SPMAPROXY) ) {
+                my $spma       = $config->getElement(SPMA)->getTree;
+                my $proxyhost = $spma->{proxyhost};
+                my $proxyport;
+                my @proxies    = split /,/, $proxyhost;
+                if ( $spma->{proxyport} ) {
+                    $proxyport = $spma->{proxyport};
+                }
+                if ($proxyhost) {
+		    if ( $repo_proxy eq 'no' ) {
+			while (@proxies) {
+			    my $prx = shift(@proxies);
+			    $prx .= ":$proxyport" if $spma->{proxyport};
+			    $url =~ s{(https?)://([^/]*)/}{$1://$prx/};
+			    $urls .= $url . ' ';
+			}
+		    } else {
+			$url .= ":$proxyport" if $spma->{proxyport};
+			$url =~ s,http?://,,;
+			$url =~ s,[^/]*/,,;
+			$url = $repo->{proxy} . $url;
+		    }
+		}
+	    }
+	}
+	if (!defined($urls)) {
+	    $urls = $url;
+	}
+        my $repofile = "[$repo->{name}]\nname=$repo->{name}\nbaseurl=$urls\nenabled=$repo->{enabled}\ngpgcheck=$repo->{gpgcheck}\n";
+        if ( defined( $repo->{mirrorlist} ) && $repo->{mirrorlist} ) {
+            $repofile = "[$repo->{name}]\nname=$repo->{name}\nmirrorlist=$url\nenabled=$repo->{enabled}\ngpgcheck=$repo->{gpgcheck}\n";
+        }
+        print $fh "# File generated by ", __PACKAGE__, ". Do not edit\n";
+        print $fh $repofile;
+        $fh->close();
+    }
+
+    # Preprocess required packages and separate version-locked
+    #    - also skip package dualities - e.g. both kernel and kernel-2.6.32-504.1.3.el6.x86_64
+    #      specified on yum install commandline will skip older version-locked package but
+    #      will install the latest what is undesired. Simply keep only version-locked variant.
+    my $pkgs               = $config->getElement(PKGS_TREE)->getTree();
+    my $wanted_pkgs        = Set::Scalar->new();
+    my $wanted_pkgs_locked = Set::Scalar->new();
+    my $found_spma         = 0;
+    my @pkl;
+    my @pkl_v;
+    my @pkl_a;
+
+    for my $name ( keys %$pkgs ) {
```

Comment:
better to call this vairbale `$escname` and do `my $name = unescape($escname);` as first thing in the loop